### ISSUE TITLE
Themed variants of images

### ISSUE THREAD

**Issue by KianNH** [readysettech/docs-old#31]
Thought I'd make an issue before opening a pull request myself to get your thoughts on a few things.

The images currently embed very nicely on the light theme, almost as if they have a transparent background.

![image](https://user-images.githubusercontent.com/94662631/198406074-b5ec24d0-1013-40c5-a389-be88d8470f8a.png)

When we swap to dark mode, you'll get the white background that's always been there.

![image](https://user-images.githubusercontent.com/94662631/198406139-38a03ad0-8032-4857-ab04-36351b13dce3.png)

The Material theme for MkDocs supports showing different variants of an image based on the state of the theme toggle [here](https://squidfunk.github.io/mkdocs-material/reference/images/#light-and-dark-mode) but that'll naturally involve a bit more maintenance burden as far as doing it retroactively and doing it in the future.

Is this something that the team is open to? Happy to contribute myself but don't want to jump into it if it's not something that you'd like to do.

As a side note, I think some of the images could benefit from zoom functionality - such as the one on the [memory efficiency](https://docs.readyset.io/concepts/efficiency/) page.

![image](https://user-images.githubusercontent.com/94662631/198406488-10d3282e-634c-4b5b-a38a-1e872b26be3d.png)

I'm not *too* sure if it supports animated images though so that's probably worth a different issue.

**Comment by jseldess** [comment]
So sorry for the delay, @KianNH. And thank you very much for the interest in improving our docs! 

Yes, I think it'd be great to customize images based on the light/dark mode state. Would very much welcome support there. In some cases, I imagine a transparent background would allow us to use a single image for both states?

**Comment by KianNH** [comment]
Yeah, I think transparent background would go a long way for the vast majority of them.

Images that have black text would still be super difficult to read on the grey background of the dark mode - vice versa if it was grey text on the white background of light mode - so they’ll want a new variant with different text.

I'll have a play with some options and post them here.
